packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,600 @@
+          autocorrect: widget.autocorrect,
+          maxLines: widget.maxLines,
+          selectionColor: _kSelectionHighlightColor,
+          selectionControls: cupertinoTextSelectionControls,
+          onChanged: widget.onChanged,
+          onEditingComplete: widget.onEditingComplete,
+          onSubmitted: widget.onSubmitted,
+          inputFormatters: formatters,
+          rendererIgnoresPointer: true,
+          cursorWidth: widget.cursorWidth,
+          cursorRadius: widget.cursorRadius,
+          cursorColor: widget.cursorColor,
+          scrollPadding: widget.scrollPadding,
+          keyboardAppearance: keyboardAppearance,
+        ),
+      ),
+    );
+
+    if (widget.placeholder != null ||
+        widget.clearButtonMode != OverlayVisibilityMode.never ||
+        widget.prefix != null ||
+        widget.suffix != null) {
+      child = ValueListenableBuilder<TextEditingValue>(
+        valueListenable: controller,
+        builder: (BuildContext context, TextEditingValue text, Widget child) {
+          final List<Widget> rowChildren = <Widget>[];
+
+          if (widget.prefix != null &&
+              _shouldAppearForModeAndText(widget.prefixMode, text.text.isNotEmpty)) {
+            rowChildren.add(widget.prefix);
+          }
+
+          final List<Widget> stackChildren = <Widget>[];
+
+          if (widget.placeholder != null && text.text.isEmpty) {
+            stackChildren.add(
+              Padding(
+                padding: widget.padding,
+                child: Text(
+                  widget.placeholder,
+                  maxLines: 1,
+                  style: widget.style.merge(
+                    const TextStyle(
+                      color: _kInactiveTextColor,
+                      fontWeight: FontWeight.w300,
+                    ),
+                  ),
+                ),
+              ),
+            );
+          }
+
+          rowChildren.add(Expanded(child: Stack(children: stackChildren..add(child))));
+
+          if (widget.suffix != null &&
+              _shouldAppearForModeAndText(widget.suffixMode, text.text.isNotEmpty)) {
+            rowChildren.add(widget.suffix);
+          } else if (_shouldAppearForModeAndText(widget.clearButtonMode, text.text.isNotEmpty)) {
+            rowChildren.add(
+              GestureDetector(
+                onTap: enabled
+                    ? () => controller.clear()
+                    : null,
+                child: const Padding(
+                  padding: EdgeInsets.symmetric(horizontal: 6.0),
+                  child: Icon(
+                    CupertinoIcons.clear_thick_circled,
+                    size: 18.0,
+                    color: _kInactiveTextColor,
+                  ),
+                ),
+              ),
+            );
+          }
+
+          return Row(children: rowChildren);
+        },
+        child: child,
+      );
+    }
+
+    return Semantics(
+      onTap: () {
+        if (!_effectiveController.selection.isValid)
```

Comment:
Would you mind wrapping this if-statement with braces?  It is especially easy to misread an implied single-line if-statement when it's sitting within a larger structure like this.

Reply:
yup. I dislike the optional braces thing too